scripts/prettier.js

```diff
@@ -0,0 +1,61 @@
+const { execSync } = require('child_process');
+const exec = require('./exec');
+const path = require('path');
+const { EOL, cpus } = require('os');
+
+const prettierIntroductionCommit = 'HEAD~1';
+const passedDiffTarget = process.argv.slice(2).length ? process.argv.slice(2)[0] : prettierIntroductionCommit;
+
+const projectPath = path.resolve(path.join(__dirname, '..'));
+const cmd = `git --no-pager diff ${passedDiffTarget} --diff-filter=AM --name-only --stat-name-width=0`;
+
+const gitDiffOutput = execSync(cmd, { cwd: projectPath });
+const filesChangedSinceLastRun = gitDiffOutput
+  .toString('utf8')
+  .split(EOL)
+  .filter(fileName => /\.(ts|tsx|js)$/.test(fileName));
+
+const prettierPath = path.resolve(__dirname, './node_modules/prettier/bin-prettier.js');
+const prettierIgnorePath = path.resolve(path.join(__dirname, '..', '.prettierignore'));
+const prettierConfigPath = path.join(
+  __dirname,
+  '..',
+  'packages',
+  'office-ui-fabric-react-tslint',
+  'prettier.config.js'
+);
+
+function runPrettierForFile(filePath) {
+  return exec(
+    `node ${prettierPath} --config ${prettierConfigPath} --ignore-path ${prettierIgnorePath} --write ${filePath}`,
```

Comment:
I understand wanting to do this in parallel, but could we just get by using this pattern?
`prettier src/**/*.js`

Or how about batching? Firing up processes like this is pretty heavy. There's an non-zero startup time cost for node.js - maybe we are better off just doing it once :)

Reply:
Good catch, actually we can do 4 processes in total instead of one per file... I overlooked the fact that it now does 1 process per file.